pyface/data_view/abstract_value_type.py

```diff
@@ -226,6 +235,76 @@ def get_image(self, model, row, column):
         from pyface.image_resource import ImageResource
         return ImageResource("image_not_found")
 
+    def has_check_state(self, model, row, column):
+        """ Whether or not the value has checked state.
+
+        The default implementation returns False.
+
+        Parameters
+        ----------
+        model : AbstractDataModel
+            The data model holding the data.
+        row : sequence of int
+            The row in the data model being queried.
+        column : sequence of int
+            The column in the data model being queried.
+
+        Returns
+        -------
+        has_check_state : bool
+            Whether or not the value has a checked state.
+        """
+        return False
+
+    def get_check_state(self, model, row, column):
+        """ The textual representation of the underlying value.
+
+        The default implementation returns "checked" if the value is
+        truthy, or "unchecked" if the value is falsey.
+
+        Parameters
+        ----------
+        model : AbstractDataModel
+            The data model holding the data.
+        row : sequence of int
+            The row in the data model being queried.
+        column : sequence of int
+            The column in the data model being queried.
+
+        Returns
+        -------
+        check_state : CheckState
+            The current checked state.
+        """
+        return (
+            CheckState.CHECKED
+            if model.get_value(row, column)
+            else CheckState.UNCHECKED
+        )
```

Comment:
I'm not sure why the default `get_check_state` doesn't return a `DataViewGetError` like `set_check_state` does.

Reply:
This is a default implementation that is almost always what you want: if the value has a check state, then it should be checked if the underlying value is truthy and unchecked if it is falsey.

By implementing it on the ABC we save repetition elsewhere (we do similar things for get text and get image, try to have a sane default for the getter).